addons/mail/static/src/core/common/message.js

```diff
@@ -84,6 +86,9 @@ export class Message extends Component {
         "thread?",
         "messageSearch?",
         "className?",
+        "avatarPlaceholder?",
+        "showDates?",
+        "isFirstMessage?"
```

Comment:
I haven't tested in practice whether this feels good or not, just a random reflexion by just looking at the code.
You can ignore it if you think it works better that way.

Looks like this props is to limit the open/close thread to the 1st message in list of a knowledge article. I don't think this is intuitive UI/UX when only 1 message has it and its the 1st in list.

In Discuss we have "thread actions", which are available in both the Discuss App in a chat window:
![Screenshot 2023-10-20 at 16 50 52](https://github.com/odoo/odoo/assets/6569390/6206d83a-ac5b-467e-882f-ba871249b968)
![Screenshot 2023-10-20 at 16 50 59](https://github.com/odoo/odoo/assets/6569390/6c28afc9-894e-4f46-8a32-4844d11a103e)

As the name suggestion, it's like message actions but on a thread. These thread actions are custom displayed by the user of `Thread` component, and in discuss this is handled by `Discuss` and `ChatWindow` components. Probably the container of `Thread` component in knowledge could display the thread actions somewhere? Doesn't mean all actions are visible, you could decide to show only some actions and up to knowledge to choose how to display it, but I think it's best to make it more readily available on `Thread` container rather than hidden in a `Message` action.

Reply:
This is a good idea, I need to discuss this with LNA to see if he would be ok with this in the v2. But I'll keep it like this for now as it was the design that was specced at the start of this project.